### DISCUSSION TITLE
How to create foreign keys?

### DISCUSSION THREAD

**Discussion by Laxystem** [komapper/komapper#1422] [Q&A] [answered]
(am using Postgres and R2DBC)

**Comment by Laxystem** [comment]
After a bit of digging, ~~found `KomapperManyToOne`~~. Flat out doesn't create any foreign keys.

**Reply by GreyTeardrop** [reply]
I believe Komapper's DDL support is intentionally designed to only be used for testing (and, partially, development), meaning it can generate DDL good enough to runs queries but any optimization or consistency features required for production (indexes, foreign keys) are not supported.

Even if that might not be convenient, I can understand why - using application models for application DB migrations are not recommended as changes to those models would result in changes to migrations that were already applied to some environments.

Disclaimer: I'm a new Komapper user so I might be wrong here, please take with a grain of salt.

**Reply by nakamura-to** [reply, marked as answer]
As @GreyTeardrop mentions, the lack of support for generating foreign keys is intentional.

See also #950.
